src/differentials.jl

```diff
@@ -191,6 +207,14 @@ end
     return element, (externed, new_state)
 end
 
+unthunk(x) = x
+unthunk(x::AbstractThunk) = unthunk(x())
+
+wirtinger_primal(::Union{AbstractThunk}) =
+    throw(ArgumentError("`wirtinger_primal` is not defined for `AbstractThunk`. Call `unthunk` first."))
+wirtinger_conjugate(::Union{AbstractThunk}) =
+    throw(ArgumentError("`wirtinger_primal` is not defined for `AbstractThunk`. Call `unthunk` first."))
```

Comment:
Will we just hit a MethodError without these? (If so, I think that would be fine)

Reply:
No, we have a fallback definition under `AbstractWirtinger`, so we definitely need these.

Reply:
cool, cool

In which case just a typo:
```suggestion
    throw(ArgumentError("`wirtinger_conjugate` is not defined for `AbstractThunk`. Call `unthunk` first."))
```